src/fides/api/alembic/migrations/versions/14acee6f5459_translation_data_clear_experience_configs.py

```diff
@@ -0,0 +1,186 @@
+"""translation data
+
+Revision ID: 14acee6f5459
+Revises: a1e23b70f2b2
+Create Date: 2024-01-09 21:17:13.115020
+
+"""
+
+import uuid
+
+import pandas as pd
+from alembic import op
+from sqlalchemy import text
+
+# revision identifiers, used by Alembic.
+
+revision = "14acee6f5459"
+down_revision = "a1e23b70f2b2"
+branch_labels = None
+depends_on = None
+
+
+def generate_record_id(prefix):
+    return prefix + "_" + str(uuid.uuid4())
+
+
+def remove_existing_experience_data(bind):
+    """
+    We remove any existing PrivacyExperience and PrivacyExperienceConfig records,
+    to allow for a "clean-slate" in creation of _new_ PrivacyExperienceConfigs based on the
+    "out of the box" templates.
+
+    This means that any existing user-edited data on PrivacyExperienceConfig records will need to be
+    _manually_ re-added to the post-migration state. It will not be migrated as part of the automatic migration.
+    """
+
+    def drop_fk_constraint(bind):
+        """
+        Drops FK constraint between PrivacyExperienceConfigHistory and PrivacyExperienceConfig,
+        to allow us to delete PrivacyExperienceConfigs without removing their associated history records
+
+        NOTE: should we just remove the column altogether? will we want to keep it around in the "new" model,
+        given that our PrivacyExperienceConfigHistory records will now be pointing to ExperienceTranslation records?
+        On the other hand, it could be useful to help in performing manual reconciliation steps post-migration.
+        """
+        op.drop_constraint(
+            "privacyexperienceconfighistory_experience_config_id_fkey",
+            "privacyexperienceconfighistory",
+            type_="foreignkey",
+        )
+
+    def delete_experiences(bind):
+        """
+        Delete all PrivacyExperience records.
+
+        These records will be effectively re-created as a result of the "out of the box" creation
+        of 'new-style' PrivacyExperienceConfig records post-migration.
+        """
+        bind.execute(
+            text(
+                """
+                DELETE FROM privacyexperience;
+                """
+            )
+        )
+
+    def delete_experience_configs(bind):
+        """
+        Delete all PrivacyExperienceConfig records.
+
+        These records will be effectively re-created via the server's "out of the box" creation
+        of the updated PrivacyExperienceConfig records, post-migration.
+        """
+        bind.execute(
+            text(
+                """
+                DELETE FROM privacyexperienceconfig;
+                """
+            )
+        )
+
+    drop_fk_constraint(bind)
+    delete_experiences(bind)
+    delete_experience_configs(bind)
```

Comment:
wouldn't that mean we should delete the experiences first? i would think that we first delete the record that is doing the pointing, rather than the one that's _pointed to_ ... but maybe i am wrong or just misunderstanding! :)

Reply:
you're right sorry i've been looking at FK's all day!